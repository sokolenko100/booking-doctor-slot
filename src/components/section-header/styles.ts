import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFFFFF',
    marginBottom: 10,
    height: 40,
    paddingHorizontal: 10,
    justifyContent: 'center',
    shadowColor: 'black',
    shadowOffset: {width: 0, height: 1},
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 2,
  },
  sectionHeader: {
    fontSize: 16,
    fontWeight: '500',
    color: '#999999',
  },
});
