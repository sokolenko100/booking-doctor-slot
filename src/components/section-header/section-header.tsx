import React, {FC, memo} from 'react';
import {Text, View} from 'react-native';
import {styles} from './styles';

interface SectionHeaderProps {
  title: string;
}

/**
 * SectionHeader is a functional component that renders a header for a section.
 * It takes a single prop, `title`, which is a string representing the title of the section.
 * The component returns a View with a Text element containing the title.
 *
 * @param {string} title - The title of the section.
 * @returns {JSX.Element} - A View containing a Text element with the title.
 */
export const SectionHeader: FC<SectionHeaderProps> = memo(({title}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.sectionHeader}>{title}</Text>
    </View>
  );
});
