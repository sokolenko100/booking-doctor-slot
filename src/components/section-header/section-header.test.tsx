import {render} from '@testing-library/react-native';
import React from 'react';
import {SectionHeader} from './section-header';

describe('SectionHeader', () => {
  it('renders the title correctly', async () => {
    const {getByText} = render(<SectionHeader title="Section Title" />);

    const sectionHeader = getByText('Section Title');
    expect(sectionHeader).toBeDefined();
  });

  it('matches the snapshot', async () => {
    const output = render(<SectionHeader title="Section Title" />);

    expect(output).toMatchSnapshot();
  });
});
