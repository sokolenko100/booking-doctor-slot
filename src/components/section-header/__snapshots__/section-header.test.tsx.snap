// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "elevation": 2,
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 10,
      "paddingHorizontal": 10,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
    }
  }
>
  <Text
    style={
      {
        "color": "#999999",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    Section Title
  </Text>
</View>
`;
